Code/GraphMol/FMCS/FMCS.h

```diff
@@ -37,13 +37,32 @@ typedef enum {
   StrictRingFusion
 } RingComparator;
 
+typedef std::map<const Atom*, std::map<const Atom*, bool>> MCSAtomDistanceCache;
+namespace FMCS {
+class RingMatchTableSet;
+}
+struct RDKIT_FMCS_EXPORT MCSCompareFunctionsData {
```

Comment:
As you see, this struct now supports passing around dynamically updated data to the standard typer functors without doing a bunch of casting of userData; userData is still available. And the python wrapper extends this structure.